scipy/differentiate/tests/test_differentiate.py

```diff
@@ -547,103 +552,113 @@ def df5(x):
     f5.mn = 3, 3  # type: ignore[attr-defined]
     f5.ref = df5  # type: ignore[attr-defined]
 
-    rosen = optimize.rosen
+    def rosen(x, _): optimize.rosen(x)
     rosen.mn = 5, 1  # type: ignore[attr-defined]
     rosen.ref = optimize.rosen_der  # type: ignore[attr-defined]
 
+    @pytest.mark.filterwarnings("ignore::UserWarning")
     @pytest.mark.parametrize('size', [(), (6,), (2, 3)])
     @pytest.mark.parametrize('func', [f1, f2, f3, f4, f5, rosen])
-    def test_examples(self, size, func):
+    def test_examples(self, size, func, xp):
+        one = xp.asarray(1.)
         rng = np.random.default_rng(458912319542)
         m, n = func.mn
         x = rng.random(size=(m,) + size)
-        res = jacobian(func, x)
-        ref = func.ref(x)
-        np.testing.assert_allclose(res.df, ref, atol=1e-10)
+        res = jacobian(lambda x: func(x , xp), xp.asarray(x, dtype=one.dtype))
+        ref = xp.asarray(np.asarray(func.ref(x)), dtype=one.dtype)
+        xp_assert_close(res.df, ref, atol=1e-10)
 
-    def test_attrs(self):
+    def test_attrs(self, xp):
         # Test attributes of result object
-        z = np.asarray([0.5, 0.25])
+        z = xp.asarray([0.5, 0.25])
 
         # case in which some elements of the Jacobian are harder
         # to calculate than others
         def df1(z):
             x, y = z
-            return [np.cos(0.5*x) * np.cos(y), np.sin(2*x) * y**2]
+            return [xp.cos(0.5*x) * xp.cos(y), xp.sin(2*x) * y**2]
 
         def df1_0xy(x, y):
-            return np.cos(0.5*x) * np.cos(y)
+            return xp.cos(0.5*x) * xp.cos(y)
 
         def df1_1xy(x, y):
-            return np.sin(2*x) * y**2
+            return xp.sin(2*x) * y**2
 
         res = jacobian(df1, z, initial_step=10)
-        assert len(np.unique(res.nit)) == 4
-        assert len(np.unique(res.nfev)) == 4
+        assert len(xp.unique_all(res.nit).values) == 4
+        assert len(xp.unique_all(res.nfev).values) == 4
 
         res00 = jacobian(lambda x: df1_0xy(x, z[1]), z[0:1], initial_step=10)
         res01 = jacobian(lambda y: df1_0xy(z[0], y), z[1:2], initial_step=10)
         res10 = jacobian(lambda x: df1_1xy(x, z[1]), z[0:1], initial_step=10)
         res11 = jacobian(lambda y: df1_1xy(z[0], y), z[1:2], initial_step=10)
         ref = optimize.OptimizeResult()
         for attr in ['success', 'status', 'df', 'nit', 'nfev']:
-            ref[attr] = np.squeeze([[getattr(res00, attr), getattr(res01, attr)],
+            ref[attr] = xp.squeeze([[getattr(res00, attr), getattr(res01, attr)],
                                     [getattr(res10, attr), getattr(res11, attr)]])
-            np.testing.assert_allclose(res[attr], ref[attr], rtol=1.5e-14)
+            xp_assert_close(res[attr], ref[attr], rtol=1.5e-14)
 
-    def test_step_direction_size(self):
+    def test_step_direction_size(self, xp):
+        one = xp.asarray(1.)
```

Comment:
If `one` is not used, why not just:
```suggestion
        dtype = xp.asarray(1.).dtype
```